labenu-system/src/endpoints/Criar/CriarDocente.ts

```diff
@@ -0,0 +1,58 @@
+import { Request, Response } from "express";
+import { v4 as generateId } from "uuid";
+import { Docente} from "../../classes/Docente";
+import connection from "../../connection";
+
+
+export async function criarDocente(req:Request, res:Response):Promise<void> {
+    let errorCode = 400
+
+    try {
+
+        const {nome, email, data_nasc, turma_id, especialidade}:Docente = req.body
+        const id = generateId()
+
+        if(!nome || !email || !data_nasc || !turma_id || !especialidade) {
+            errorCode = 400
+            res.send("Todos os campos são obrigatórios.")
+        }
+
+        if (typeof(especialidade) != "object") {
+            res.status(errorCode).send("A(s) especialidade(s) deve(m) vir em forma de array!")
+            return
+        }
```

Comment:
Isso aqui não garante que a variável é uma array, pois ela pode ser um objeto e passar nessa verificação. Uma solução seria checar, também, se a propriedade length é diferente de undefined. Se for, vai ser uma array e não um objeto.